domain/consensus/utils/merkle/merkle.go

```diff
@@ -90,12 +82,12 @@ func merkleRoot(hashes []*externalapi.DomainHash) *externalapi.DomainHash {
 			merkles[offset] = nil
 
 		// When there is no right child, the parent is generated by
-		// hashing the concatenation of the left child with itself.
+		// hashing the concatenation of the left child with zeros.
```

Comment:
What are the benefits of this over concatenating the child with itself?

Reply:
This prevents having 2 separate inclusion proofs for the same thing in the same merkle tree

Reply:
And it is also the standard for how merkle trees are built, see: https://tools.ietf.org/html/rfc6962#section-2.1